### DISCUSSION TITLE
Argocd AppofApps Out of sync

### DISCUSSION THREAD

**Discussion by saikumar13** [argoproj/argo-cd#9521] [General]
Hi Everyone !
I am facing issue with argocd app of apps though we are maintaining separate repos for each application component and we have defined separate argocd application object as well initially it shows all the applications are in sync but intermittently few components/apps reporting out of sync even though there is no changes were made to the repo. 

Could you please suggest on this.

**Comment by jgwest** [comment]
Hi @saikumar13 , you should be able to examine the Applications within the Argo CD UI to determine the specific differences between what is defined in Git, and what is defined on the actual cluster.

In general, when Applications report as out of sync, and there were no changes to the Git repository, it is due to changes to the K8s cluster resources themselves. Examining the diffs should tell you what changed. More information on [Argo CD Diffing is here](https://argo-cd.readthedocs.io/en/stable/user-guide/diffing/).

One option is to enable [selfHeal](https://argo-cd.readthedocs.io/en/stable/user-guide/auto_sync/#automatic-self-healing) on the Application, which will cause Argo CD to attempt to "fix" any deviation of those resources from what is defined in Git.
